### PR TITLE
[IDM-88] Add support for Kong JWT consumers

### PR DESCRIPTION
**Pull request by janslow**
Converts this:
```yaml
# shipcat.conf
regions:
- kong:
    jwt_consumers:
      my-consumer:
        issuer: https://my-issuer/
        public_key: |-
          -----BEGIN PUBLIC KEY-----
          ...
          -----END PUBLIC KEY-----
```

Into this:
```yaml
# Output of `shipcat kong`
consumers:
  - username: my-consumer
    credentials:
      - name: jwt
        attributes:
          algorithm: RS256
          key: https://my-issuer/
          rsa_public_key: |-
            -----BEGIN PUBLIC KEY-----
            ...
            -----END PUBLIC KEY-----
```

(part of #60)

# Testing
I've compared the output of `shipcat kong` (0.81.0) using  with `cargo run kong` against `master` of the manifests repo, which makes no changes.

Also see https://github.com/Babylonpartners/manifests/pull/4675 for an example of a JWT consumer and the Kong diff.

### CODE REVIEW

File: shipcat_definitions/src/structs/kongfig.rs [hunk 1]
```diff
@@ -503,21 +512,29 @@ pub struct Consumer {
 }
 
 #[derive(Serialize, Deserialize, Clone)]
-#[serde(deny_unknown_fields)]
-pub struct ConsumerCredentials {
-    pub name: String,
-    pub attributes: ConsumerCredentialsAttributes,
+#[serde(tag = "name", content = "attributes", rename_all="kebab-case")]
```
Comment (clux) [inline shipcat_definitions/src/structs/kongfig.rs]: ooh, this is a fun one. nice.

File: shipcat_definitions/src/structs/kongfig.rs [hunk 2]
```diff
@@ -472,18 +472,27 @@ pub fn kongfig_consumers(k: KongConfig) -> Vec<Consumer> {
...
+        Consumer {
+            username: k.to_string(),
+            acls: vec![],
+            credentials: vec![ConsumerCredentials::Jwt(JwtCredentialsAttributes {
+                key: v.issuer,
+                algorithm: "RS256".into(),
+                rsa_public_key: v.public_key,
+            })]
+        }
+    }).for_each(|c| consumers.push(c));
```
Comment (clux) [inline shipcat_definitions/src/structs/kongfig.rs]: nice, clarifies that consumers are being pushed to from two places.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
+root = true
```
Comment (janslow) [inline .editorconfig]: FYI, [`.editorconfig`](https://editorconfig.org/) is a cross language/IDE way to specify code format rules.

It should stop you needing to remind me to add final newlines :P

Comment (clux) [reply .editorconfig]: yeah, it's a good one to have :+1:

**Comment by janslow** [comment]
This is ready to merge (I can bump the version if you want, or do it in #81)

**Comment by clux** [comment]
Feel free to bump in the next one if you don't need this immediately propagated. There's a bump versions script in scripts/ to ensure versions stay in sync.

**Pull request merged**